teps/0029-step-workspaces.md

```diff
@@ -101,3 +235,195 @@ to other `Steps` in the same Task performing ancillary work.
 As a Pipeline author I want to be able to quickly audit that the `git-fetch` Task I am using in
 my Pipeline is only exposing the git SSH key for my team's source repo in the single `Step` that
 performs `git clone`, and not to any `Steps` in the same Task performing ancillary work.
+
+## Design Details
+
+Add a new `WorkspaceUsage` struct to the `task_types.go` file:
+
+```go
+// WorkspaceUsage declares that a Step or Sidecar utilizes a Task's Workspace.
+type WorkspaceUsage struct {
+  // Name is the name of the Task's WorkspaceDeclaration that this Step or Sidecar is utilizing. It is required.
+  Name  string `json:"name"`
```

Comment:
nit: according to latest updates, this should hold the `mountPath` field too 😉

Reply:
d'oh!  fixed!